File: src/OneDOF/Abutment/Abutment/Abutment.java
package OneDOF.Abutment.Abutment;

import java.awt.BorderLayout;
import java.awt.Image;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.Date;

import javax.swing.JApplet;
import javax.swing.JButton;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.Timer;

@SuppressWarnings("serial")
public class Abutment extends JApplet implements ActionListener {
   static final double kMinWn = 0.1;
   static final double kMaxWn = 2.0;
   static final double kMinXi = 0.001;
   static final double kMaxXi = 2.0;
   static final double kMinXin = -17.0;
   static final double kMaxXin = 17.0;
   static final double kMinPulsewidth = 0.001;
   static final double kMaxPulsewidth = 10.0;
   static final String kRunString = "Run";
   CFramePanel mFramePanel;
   CFrameAnimation mAnimFrame;
   Timer mTimer = null;
   long mLastTime = 0;

   public void init() {
      this.setLayout(new BorderLayout());
      JPanel buttonPanel = new JPanel();
      this.add("North", buttonPanel);
      JLabel copyright = new JLabel("This program is copyright, you need permission to use it. © 1999 B J Stone, bjs@mech.uwa.edu.au.");
      this.add("South", copyright);
      this.mFramePanel = new CFramePanel(this);
      this.add("Center", this.mFramePanel);
      this.mAnimFrame = new CFrameAnimation(this.mFramePanel, 0, 0, this.getSize().width, this.getSize().height);
      CFrameSmallControl WnControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 2, 10, 10, 0.1, 1.5, 2.0, "Hz");
      WnControl.SetLabelOffset(20, 10);
      WnControl.SetTextEditable(true);
      Image wnImage = this.getImage(this.getCodeBase(), "wn.gif");
      new CFramePicture(this.mFramePanel, 35, 26, 0, 0, wnImage, false);
      CFrameSmallControl XiControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 3, 100, 10, 0.001, 0.05, 2.0, " ");
      XiControl.SetLabelOffset(20, 10);
      XiControl.SetTextEditable(true);
      Image XiImage = this.getImage(this.getCodeBase(), "xi.gif");
      new CFramePicture(this.mFramePanel, 128, 25, 0, 0, XiImage, false);
      CFrameSmallControl XinControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 4, 190, 10, -17.0, 7.0, 17.0, "mm");
      XinControl.SetLabelOffset(20, 10);
      XinControl.SetTextEditable(true);
      Image XinImage = this.getImage(this.getCodeBase(), "Xin.gif");
      new CFramePicture(this.mFramePanel, 214, 25, 0, 0, XinImage, false);
      CFrameSmallControl PulsewidthControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 5, 280, 10, 0.001, 2.0, 10.0, "secs");
      PulsewidthControl.SetLabelOffset(20, 10);
      PulsewidthControl.SetTextEditable(true);
      Image PulsewidthImage = this.getImage(this.getCodeBase(), "Pulsewidth.gif");
      new CFramePicture(this.mFramePanel, 304, 25, 0, 0, PulsewidthImage, false);
      new CFrameHelp(this.mFramePanel, 0, 0, 80, 80, "This is the natural frequency\nof the system.");
      new CFrameHelp(this.mFramePanel, 90, 0, 80, 80, "This is the damping\nratio for the system.");
      new CFrameHelp(this.mFramePanel, 180, 0, 80, 80, "This is the magnitude of \nthe input step.");
      new CFrameHelp(this.mFramePanel, 270, 0, 80, 80, "This is duration of the pulse");
      this.mAnimFrame.mStartButton = new JButton("Start");
      this.mAnimFrame.mStartButton.addActionListener(this);
      buttonPanel.add(this.mAnimFrame.mStartButton);
      this.mAnimFrame.mPauseButton = new JButton("Pause");
      this.mAnimFrame.mPauseButton.addActionListener(this);
      buttonPanel.add(this.mAnimFrame.mPauseButton);
      this.mAnimFrame.mStopButton = new JButton("Stop");
      this.mAnimFrame.mStopButton.addActionListener(this);
      buttonPanel.add(this.mAnimFrame.mStopButton);
      this.validate();
      this.mAnimFrame.UpdateButtonAppearance();
   }

   public void start() {
      this.mTimer = new Timer(20, this);
      this.mTimer.start();
   }

   public void actionPerformed(ActionEvent e) {
      // Handle button clicks
      if (e.getSource() instanceof JButton) {
         JButton source = (JButton) e.getSource();
         String label = source.getText();
         if (label.equals("Start") || label.equals("Cont.")) {
            this.mAnimFrame.ControlMessage(1, 1.0);
         } else if (label.equals("Pause")) {
            this.mAnimFrame.ControlMessage(1, 2.0);
         } else if (label.equals("Stop")) {
            this.mAnimFrame.ControlMessage(1, 0.0);
         }
         return;
      }

      // Handle timer events
      if (this.mTimer != null) {
         Date ddd = new Date();
         long thisTime = ddd.getTime();

         if (this.mLastTime == 0) {
            this.mLastTime = thisTime;
         }
         double realSeconds = (thisTime - this.mLastTime) / 1000.0;
         this.mLastTime = thisTime;

         this.mAnimFrame.ControlMessage(0, realSeconds);
      }
   }

   public void stop() {
      if (this.mTimer != null) { this.mTimer.stop(); this.mTimer = null; }
   }
}

File: site/swingjs/j2s/OneDOF/Abutment/Abutment/Abutment.js
(function(){var P$=Clazz.newPackage("OneDOF.Abutment.Abutment"),I$=[[0,'java.awt.BorderLayout','java.awt.Panel','java.awt.Label','OneDOF.Abutment.Abutment.CFramePanel','OneDOF.Abutment.Abutment.CFrameAnimation','OneDOF.Abutment.Abutment.CFrameSmallControl','OneDOF.Abutment.Abutment.CFramePicture','OneDOF.Abutment.Abutment.CFrameHelp','java.awt.Button','javax.swing.Timer']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Abutment", null, 'java.applet.Applet', 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.mTimer=null;
this.mLastTime=0;
},1);

C$.$fields$=[['J',['mLastTime'],'O',['mFramePanel','OneDOF.Abutment.Abutment.CFramePanel','mAnimFrame','OneDOF.Abutment.Abutment.CFrameAnimation','mTimer','javax.swing.Timer']]]

Clazz.newMeth(C$, ['init$','init'],  function () {
this.setLayout$java_awt_LayoutManager(Clazz.new_($I$(1,1)));
var buttonPanel=Clazz.new_($I$(2,1));
this.add$S$java_awt_Component("North", buttonPanel);
var copyright=Clazz.new_($I$(3,1).c$$S,["This program is copyright, you need permission to use it. \u00a9 1999 B J Stone, bjs@mech.uwa.edu.au."]);
this.add$S$java_awt_Component("South", copyright);
this.mFramePanel=Clazz.new_($I$(4,1).c$$java_applet_Applet,[this]);
this.add$S$java_awt_Component("Center", this.mFramePanel);
this.mAnimFrame=Clazz.new_([this.mFramePanel, 0, 0, this.getSize$().width, this.getSize$().height],$I$(5,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I);
var WnControl=Clazz.new_($I$(6,1).c$$OneDOF_Abutment_Abutment_CFramePanel$OneDOF_Abutment_Abutment_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 2, 10, 10, 0.1, 1.5, 2.0, "Hz"]);
WnControl.SetLabelOffset$I$I(20, 10);
WnControl.SetTextEditable$Z(true);
var wnImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "wn.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 35, 26, 0, 0, wnImage, false]);
var XiControl=Clazz.new_($I$(6,1).c$$OneDOF_Abutment_Abutment_CFramePanel$OneDOF_Abutment_Abutment_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 3, 100, 10, 0.001, 0.05, 2.0, " "]);
XiControl.SetLabelOffset$I$I(20, 10);
XiControl.SetTextEditable$Z(true);
var XiImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "xi.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 128, 25, 0, 0, XiImage, false]);
var XinControl=Clazz.new_($I$(6,1).c$$OneDOF_Abutment_Abutment_CFramePanel$OneDOF_Abutment_Abutment_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 4, 190, 10, -17.0, 7.0, 17.0, "mm"]);
XinControl.SetLabelOffset$I$I(20, 10);
XinControl.SetTextEditable$Z(true);
var XinImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Xin.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 214, 25, 0, 0, XinImage, false]);
var PulsewidthControl=Clazz.new_($I$(6,1).c$$OneDOF_Abutment_Abutment_CFramePanel$OneDOF_Abutment_Abutment_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 5, 280, 10, 0.001, 2.0, 10.0, "secs"]);
PulsewidthControl.SetLabelOffset$I$I(20, 10);
PulsewidthControl.SetTextEditable$Z(true);
var PulsewidthImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Pulsewidth.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 304, 25, 0, 0, PulsewidthImage, false]);
Clazz.new_($I$(8,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$S,[this.mFramePanel, 0, 0, 80, 80, "This is the natural frequency\nof the system."]);
Clazz.new_($I$(8,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$S,[this.mFramePanel, 90, 0, 80, 80, "This is the damping\nratio for the system."]);
Clazz.new_($I$(8,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$S,[this.mFramePanel, 180, 0, 80, 80, "This is the magnitude of \nthe input step."]);
Clazz.new_($I$(8,1).c$$OneDOF_Abutment_Abutment_CFramePanel$I$I$I$I$S,[this.mFramePanel, 270, 0, 80, 80, "This is duration of the pulse"]);
this.mAnimFrame.mStartButton=Clazz.new_($I$(9,1).c$$S,["Start"]);
this.mAnimFrame.mStartButton.addActionListener$java_awt_event_ActionListener(this);
buttonPanel.add$java_awt_Component(this.mAnimFrame.mStartButton);
this.mAnimFrame.mPauseButton=Clazz.new_($I$(9,1).c$$S,["Pause"]);
this.mAnimFrame.mPauseButton.addActionListener$java_awt_event_ActionListener(this);
buttonPanel.add$java_awt_Component(this.mAnimFrame.mPauseButton);
this.mAnimFrame.mStopButton=Clazz.new_($I$(9,1).c$$S,["Stop"]);
this.mAnimFrame.mStopButton.addActionListener$java_awt_event_ActionListener(this);
buttonPanel.add$java_awt_Component(this.mAnimFrame.mStopButton);
this.validate$();
this.mAnimFrame.UpdateButtonAppearance$();
});

Clazz.newMeth(C$, ['start$','start'],  function () {
this.mTimer=Clazz.new_($I$(10,1).c$$I$java_awt_event_ActionListener,[20, this]);
this.mTimer.start$();
});

Clazz.newMeth(C$, ['actionPerformed$java_awt_event_ActionEvent','actionPerformed'],  function (e) {
if (Clazz.instanceOf(e.getSource$(), "java.awt.Button")) {
var source=e.getSource$();
var label=source.getLabel$();
if (label.equals$O("Start") || label.equals$O("Cont.") ) {
this.mAnimFrame.ControlMessage$I$D(1, 1.0);
} else if (label.equals$O("Pause")) {
this.mAnimFrame.ControlMessage$I$D(1, 2.0);
} else if (label.equals$O("Stop")) {
this.mAnimFrame.ControlMessage$I$D(1, 0.0);
}return;
}if (this.mTimer != null ) {
var ddd=Clazz.new_(java.util.Date);
var thisTime=ddd.getTime$();
if (Long.$eq(this.mLastTime,0 )) {
this.mLastTime=thisTime;
}var realSeconds=(Long.$sub(thisTime,this.mLastTime)) / 1000.0;
this.mLastTime=thisTime;
this.mAnimFrame.ControlMessage$I$D(0, realSeconds);
}});

Clazz.newMeth(C$, ['stop$','stop'],  function () {
if (this.mTimer != null ) {
this.mTimer.stop$();
this.mTimer=null;
}});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:50 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
